services/wallet/transfer/database.go

```diff
@@ -503,18 +505,20 @@ func updateOrInsertTransfersDBFields(creator statementCreator, transfers []trans
 }
 
 func removeGasOnlyEthTransfer(creator statementCreator, t transferDBFields) error {
-	if t.transferType != w_common.EthTransfer {
-		query, err := creator.Prepare(`DELETE FROM transfers WHERE tx_hash = ? AND address = ? AND network_id = ?
-		 AND account_nonce = ? AND type = 'eth' AND amount_padded128hex = '00000000000000000000000000000000'`)
-		if err != nil {
-			return err
-		}
+	query, err := creator.Prepare(`DELETE FROM transfers WHERE tx_hash = ? AND address = ? AND network_id = ? AND account_nonce = ? AND type = 'eth' AND amount_padded128hex = '00000000000000000000000000000000'`)
```

Comment:
how do we distinguish intentional 0-valued transfers from these gas-only ones?

Reply:
that's the problem, we don't i think. I had another solution:
`query, err := creator.Prepare(`DELETE FROM transfers WHERE type = 'eth' AND EXISTS (SELECT 1 FROM transfers AS t2 WHERE transfers.tx_hash = t2.tx_hash AND transfers.type != t2.type)`)
`
But Ivan had performance concerns

Reply:
Added a count check guard before @dlipicar @IvanBelyakoff

Reply:
@IvanBelyakoff 
Before deleting any rows, i am now making sure that there is at least more than 1 tx with same tx hash.
Which means that if there is a single row for 0 eth transfer, there will be one row and i won't delete it no?